src/lib/web3-instance.ts

```diff
@@ -1,14 +1,15 @@
+import { ChainId } from '@shapeshiftoss/caip'
 import Web3 from 'web3'
-import { getWeb3Provider } from 'lib/web3-provider'
+import { getWeb3ProviderByChainId } from 'lib/web3-provider'
 
-let maybeWeb3Instance: Web3 | null
+const web3InstanceMap: Map<ChainId, Web3> = new Map()
 
-export const getWeb3Instance = (): Web3 => {
-  const web3Provider = getWeb3Provider()
-  if (!maybeWeb3Instance) {
-    maybeWeb3Instance = new Web3(web3Provider)
-    return maybeWeb3Instance!
+export const getWeb3Instance = (chainId: ChainId): Web3 => {
```

Comment:
`getWeb3InstanceByChainId`

Reply:
Good spot, addressed in https://github.com/shapeshift/web/pull/2164/commits/57f023bf9cb13475460440a417d188e49824bae4